File: backend/Models/Entities/PetReport.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Home4Paws.API.Models.Entities
{
    public class PetReport
    {
        public Guid Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Type { get; set; } // Cat, Dog, Other

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [MaxLength(100)]
        public string Breed { get; set; }

        [Required]
        [MaxLength(100)]
        public string Color { get; set; }
        
        [Required]
        [MaxLength(20)]
        public string Age { get; set; }  // Puppy, Young, Adult, Senior

        [Required]
        [MaxLength(20)]
        public string Gender { get; set; } // Male, Female

        [Required]
        [MaxLength(100)]
        public string Size { get; set; } // Small, Medium, Large

        [MaxLength(1000)]
        public string Description { get; set; }

        [Required]
        [MaxLength(20)]
        public string ReportType { get; set; } // Lost or Found

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "Pending Confirmation";

        public DateTime DateReported { get; set; } = DateTime.UtcNow;

        [Required]
        public DateTime LostOrFoundDate { get; set; }

        [Required]
        [MaxLength(200)]
        public string Location { get; set; }

        public double? Latitude { get; set; }
        public double? Longitude { get; set; }

        [Required]
        [MaxLength(100)]
        [RegularExpression(@"^[a-zA-Z\s]*$", ErrorMessage = "Contact name can only contain letters and spaces")]
        public string ContactName { get; set; }

        [Required]
        [MaxLength(10)]
        [MinLength(10)]
        [RegularExpression(@"^\d{10}$", ErrorMessage = "Phone number must be exactly 10 digits")]
        public string Phone { get; set; }

        [Required]
        [MaxLength(255)]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        public string[] PhotoUrls { get; set; }

        [MaxLength(1000)]
        public string IdentifyingFeatures { get; set; } // Distinctive marks, collar details, etc.

        [MaxLength(500)]
        public string MedicalConditions { get; set; } // Any known medical conditions

        public bool IsChipped { get; set; }
        public string ChipNumber { get; set; }

        public bool HasReward { get; set; }
        [MaxLength(20)]
        public string RewardAmount { get; set; }

        public int Views { get; set; } = 0;
        public bool IsUrgent { get; set; }
        public bool IsClosed { get; set; }
        public DateTime? ClosedAt { get; set; }
        public string ClosureReason { get; set; } // Found, Reunited, etc.

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        [MaxLength(500)]
        public string AdminNotes { get; set; }
    }
}

File: frontend/src/types/petReport.ts
/**
 * TypeScript interfaces for Pet Reports matching backend DTOs
 */

export interface PetFinderDto {
  id: string; // Guid as string from backend
  type: 'lost' | 'found';
  petName: string;
  description: string;
  imageUrl: string;
  allImages: string[];
  contactInfo: string; // Anonymized contact info
  location: string;
  dateReported: string; // ISO date string
  lostOrFoundDate: string; // ISO date string
  breed: string;
  color: string;
  size: string;
  age: string;
  gender: string;
  hasReward: boolean;
  rewardAmount: string;
  isUrgent: boolean;
  views: number;
}

export interface UserPetReportDto {
  id: string; // Guid as string from backend
  type: 'lost' | 'found';
  petName: string;
  description: string;
  imageUrl: string;
  status: 'pending' | 'active' | 'resolved' | 'archived';
  dateReported: string; // ISO date string
  lostOrFoundDate: string; // ISO date string
  location: string;
  views: number;
  updatedAt?: string; // ISO date string
  adminNotes?: string;
}

export interface AdminPetReportDto {
  id: string; // Guid as string from backend
  type: 'lost' | 'found';
  petName: string;
  description: string;
  imageUrl: string;
  status: 'pending' | 'active' | 'resolved' | 'archived';
  dateReported: string; // ISO date string
  location: string;
  contactName: string;
  phone: string;
  email: string;
  isUrgent: boolean;
  views: number;
  updatedAt?: string; // ISO date string
  adminNotes?: string;
}

export interface UpdateReportStatusRequest {
  status: 'pending' | 'active' | 'resolved' | 'archived';
  adminNotes?: string;
}